### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by 3akev**

### CODE REVIEW

File: src/Scripts/Ecs/Extension/EntityFactory.gd [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+func create_player(tile):
+    return Entity.new(tile.x, tile.y) \
+		.add("SpriteComponent", SpriteComponent.new("Creatures", "Player")) \
+		.add("PlayerMovementComponent", PlayerMovementComponent.new()) \
+		.add("CameraFollowComponent", CameraFollowComponent.new()) \
+		.add("AttackComponent", AttackComponent.new(Constants.PLAYER_ATTACK_DAMAGE)) \
+		.add("HealthComponent", HealthComponent.new(Constants.PLAYER_MAX_HEALTH)) \
+        .add("SightComponent", SightComponent.new(Constants.PLAYER_SIGHT))
+
+func create_monster(tile):
```
Comment (nightblade9) [inline src/Scripts/Ecs/Extension/EntityFactory.gd]: Please change these methods to take in `(x, y)` instead of tile. Tile doesn't make sense as something the entity factory should know about.

File: src/Scripts/Ecs/Systems/ChasePlayerSystem.gd [hunk 2]
```diff
@@ -25,10 +25,12 @@ func on_turn_end():
 		if distance <= entity.get("SightComponent").sight_radius:
 			
 			var direction = entity.position.direction_to(_player.position)
+			var mod = Vector2()
```
Comment (nightblade9) [inline src/Scripts/Ecs/Systems/ChasePlayerSystem.gd]: This should really be called `diff` instead of `mod`. Mod makes me think "modulo".

File: src/Scripts/Ecs/Systems/DisplaySystem.gd [hunk 3]
```diff
@@ -4,11 +4,10 @@ var _tiles_tilemap:TileMap
 var _entities_tilemap:TileMap
 
 var tilemaps_by_name = {}
-var _fov_cache = {} # [x, y] => true/false
-var _fog_of_war_seen_tiles = {} # [x, y] instances for O(1) search
```
Comment (nightblade9) [inline src/Scripts/Ecs/Systems/DisplaySystem.gd:8]: Why did you remove these extremely helpful comments? I has a sad. (I added them because it's not obvious why we're not using a simple list/array here)

File: src/Scripts/Ecs/Systems/DisplaySystem.gd [hunk 4]
```diff
@@ -21,68 +20,38 @@ func _init(event_bus, tiles_tilemap:TileMap, entities_tilemap:TileMap):
...
-			if not _fov_cache[key]:
-				if _fog_of_war_seen_tiles.has(key):
-					_tiles_tilemap.set_cell(x, y, _tiles_tilemap.tile_set.find_tile_by_name("DiscoveredFloor"))
-				else:
-					_tiles_tilemap.set_cell(x, y, _tiles_tilemap.tile_set.find_tile_by_name("Fog"))
-			else:
+			var pos = Vector2(x, y)
+			
+			# flood with walls
+			_tiles_tilemap.set_cell(x, y, _tiles_tilemap.tile_set.find_tile_by_name("Fog"))
```
Comment (nightblade9) [inline src/Scripts/Ecs/Systems/DisplaySystem.gd:34]: Comment says "flood with walls" but you're flooding it with fog?!

File: src/Scripts/Ecs/Systems/FovSystem.gd [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+var _event_bus
+var _player = null
+
+func _init(event_bus):
+	_event_bus = event_bus
+	
+	_event_bus.connect("spawn_entity", self, "on_spawn_entity")
+	_event_bus.connect("move_entity", self, "on_move_entity")
+
+func on_spawn_entity(entity):
```
Comment (nightblade9) [inline src/Scripts/Ecs/Systems/FovSystem.gd:12]: We use this pattern of "hook into spawn_entity to find the player" a few places, I wonder if we should instead just pass the player through the system constructor or do something else? It seems a big ... eh, excessive.

**Review by nightblade9 — CHANGES_REQUESTED** [review]
LGTM, needs a couple of tweaks though.

File: src/Scripts/Ecs/Systems/DisplaySystem.gd (hunk 3 above)
Comment (3akev) [reply src/Scripts/Ecs/Systems/DisplaySystem.gd:8]: They got lost between iterations and copypasting. The first one is outdated, with the new FOV system. I'll bring back the fog of war comment, but update it to reference Vector2 which we now use as keys.

File: src/Scripts/Ecs/Systems/FovSystem.gd (hunk 5 above)
Comment (3akev) [reply src/Scripts/Ecs/Systems/FovSystem.gd:12]: We can have a designated signal!

Sometimes I feel like I'm overusing signals.
